lib/elixir/lib/registry.ex

```diff
@@ -444,6 +444,11 @@ defmodule Registry do
   value or tuple element, while :"$1" can be used to temporarily assign part
   of pattern to a variable for a subsequent comparison.
 
+  It is possible to pass list of guard conditions for more precise matching.
+  Each guard is a tuple, which describes check that should be passed by assigned part of pattern.
+  For example :"$1" > 1 guard condition would be expressed as {:>, :"$1", 1} tuple.
+  Please note that guard conditions will work only for assigned variables like :"$1", :"$2", etc.
```

Comment:
Done.
I played around a little bit with it - it seems like in some cases you can use :"$_ in guard clauses without having assigned variables.

Example:
```
iex(1)> Registry.start_link(:duplicate, Registry.MatchTest)
{:ok, #PID<0.85.0>}
iex(2)> {:ok, _} = Registry.register(Registry.MatchTest, "hello", {1, :atom, 1})
{:ok, #PID<0.86.0>}
iex(3)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :"$1"}, [{:is_tuple, :"$_"}]) |> Enum.sort()
[{#PID<0.81.0>, {1, :atom, 1}}]
iex(4)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :"$1"}, [{:is_tuple, :"$$"}]) |> Enum.sort()
[]
iex(5)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :_}, [{:is_tuple, :"$_"}]) |> Enum.sort()   
[{#PID<0.81.0>, {1, :atom, 1}}]
iex(6)> Registry.match(Registry.MatchTest, "hello", {:_, :_, :_}, [{:is_tuple, :"$$"}]) |> Enum.sort()
[]
```